tests/BeatmapsetDisqualifyTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace Tests;
+
+use App\Events\NewPrivateNotificationEvent;
+use App\Jobs\BroadcastNotification;
+use App\Models\Beatmap;
+use App\Models\Beatmapset;
+use App\Models\Notification;
+use App\Models\User;
+use App\Models\UserNotificationOption;
+use Event;
+use Queue;
+
+class BeatmapsetDisqualifyTest extends TestCase
+{
+    /** @var Beatmapset */
+    protected $beatmapset;
+
+    /** @var User */
+    protected $sender;
+
+    /** @var User */
+    protected $user;
+
+    public function testDuplicateNotificationNotSent()
+    {
+        $this->beatmapset->watches()->create(['user_id' => $this->user->getKey()]);
+        $this->createNotificationOption();
+
+        $this->disqualify()->assertStatus(200);
+
+        Queue::assertPushed(BroadcastNotification::class, function (BroadcastNotification $job) {
+            return optional($job->handle())->name === Notification::BEATMAPSET_DISQUALIFY;
+        });
+
+        $dispatchedCount = 0;
+        Event::assertDispatched(NewPrivateNotificationEvent::class, function (NewPrivateNotificationEvent $event) use (&$dispatchedCount) {
+            if ($event->notification->name === Notification::BEATMAPSET_DISQUALIFY) {
+                $this->assertSame(array_unique($event->getReceiverIds()), $event->getReceiverIds());
+                $dispatchedCount++;
+            }
+
+            return true;
+        });
+
+        $this->assertSame(1, $dispatchedCount);
```

Comment:
```
$events = Event::dispatched(...::class, function (...) {
  if (...) {
    $this->assertSame(...);

    return true;
  }

  return false;
});

$this->assertSame(1, $events->count());
```

`assertDispatched` is literally above just `> 0` instead of `=== 1`.